lib/loggingService/LogEntry.js

```diff
@@ -110,68 +165,13 @@ function LogEntry(channel) {
 }
 
 
-LogEntry.prototype.registerErrorDetails = function (error) {
-	// the stack trace and other error information will be moved into the data object
-
-	// turn the stack string into an array, stripped from its noisy whitespace
-
-	var stack = error.stack && error.stack.split(/\s*\n\s*at\s+/);
-	if (!stack || stack.length <= 1) {
-		return;
-	}
-
-	// remove the first line: "Error: name", leaving only stack frames
-
-	stack.shift();
-
-	this.data = this.data || {};
-
-	// assign all the data to the data object (custom errors may have some)
-	Object.assign(this.data, error);
-
-	// Remove message, we already display it in the log text
-	delete this.data.message;
-
-	// reformat the stack
-
-	var locationFound = false;
-
-	for (var i = 0; i < stack.length; i += 1) {
-		// make the file path relative to the project
-
-		stack[i] = stack[i].replace(rootPath + '/', '');
-
-		// If the error did not originate in a native function (like JSON.parse), the top stack
-		// frame should carry file, line and character offset information. While we're looking
-		// at a native function, we move on to the next stack frame to get us closest to the
-		// source of the error in user-land code.
-
-		if (!locationFound) {
-			var m = stack[i].match(/([\w\.]+):([0-9]+):([0-9]+)/);
-			if (m) {
-				locationFound = true;
-
-				this.data.file = m[1];
-				this.data.line = parseInt(m[2], 10) || 0;
-				this.data.offset = parseInt(m[3], 10) || 0;
-			}
-		}
-	}
-
-	if (error.code) {
-		this.data.code = error.code;
-	}
-
-	this.data.type = error.name;
-	this.data.stack = stack;
-};
-
-
 LogEntry.prototype.serializeArgument = function (arg) {
 	if (arg instanceof Error) {
-		this.registerErrorDetails(arg);
-
+		const error = arg;
 		arg = arg.message;
+
+		delete error.message;
```

Comment:
I'm not a fan of making a deep copy of error objects. The `delete` could happen on the return value of `serializeError`.